lib/utilityservice.js

```diff
@@ -1359,4 +1359,18 @@ UtilityService.prototype.getVisualForcePageUrl = function (dataPackType, datapac
         designerPath = '/' + datapackId;
     }
     return designerPath;
+}
+
+UtilityService.prototype.fetchAllRecords =  async function(query){
+    try {
+        var fetchedRecords = [];
+        var result = await this.vlocity.jsForceConnection.tooling.query(query);
+        fetchedRecords = fetchedRecords.concat(result.records);
+        if (result.records.length === 2000) {
+            fetchedRecords = fetchedRecords.concat(await this.fetchAllRecords(query + ' OFFSET 2000'));
```

Comment:
Did you test that offset works here?

Reply:
Yes. It worked here as in my org it has 2568 ApexClasses, So it fetched in sets of 2000 & 568